### DISCUSSION TITLE
Suspend authenticated users

### DISCUSSION THREAD

**Discussion by magnusskare** [pocketbase/pocketbase#4420] [Q&A] [answered]
I currently have a login function where the user has to authenticate with a third party service.

```
function login() {
    const w = window.open();
    if (!w) return false;

    (async function() {
      await pb.collection("users").authWithOAuth2({
        provider: 'microsoft',
        urlCallback: (url) => {
            w.location.href = url
        },
      })
    })();
  }
```

This works well (even with Safari), but if I delete their user, as the application is already authenticated, they're right back in. As this is the case, I would prefer to be able to suspend their user, but keep them in the user collection.

Is there a clean way to do this built-in? Would I have to subscribe to realtime updates to log them out, or can this be done globally in the entire application. I'm using PocketBase out of the box with PocketHost.

I'm a bit of a novice to all of this, so any keywords or topics worth researching would be appreciated. I've read [this topic](https://github.com/pocketbase/pocketbase/discussions/2168) on this already, but I'm not using PocketBase as a framework (I think).

**Comment by ganigeorgiev** [comment, marked as answer]
_See also https://github.com/pocketbase/pocketbase/discussions/4396._

There is no builtin mechanism to suspend/deactivate a user at the moment.

As mentioned in the linked discussions the usual options are to add a `bool` or `select` status field and check its value from the API rules for each collection or via the hooks. Here is one possible example using a global [JS `pb_hooks` middleware](https://pocketbase.io/docs/js-routing/#custom-middlewares):

```js
// pb_hooks/main.pb.js

routerUse((next) => {
    return (c) => {
        // retrieve the current authenticated user (if any)
        const user = c.get("authRecord");

        // check its "active" bool field
        //
        // note: make sure that users can't update the "active" value
        // on their own by defining a users Update API rule like
        // `@request.data.active:isset = false` or similar
        if (user && !user.get("active")) {
            // depending on the desired behaviour you can also make the error conditional and throw only for specific paths
            throw new ForbiddenError("You are not allowed to access the requested resource.")
        }

        return next(c)
    }
})
```

**Reply by magnusskare** [reply]
Thank you, this is very helpful. I got a solution working by denying API access globally.

I'll do some reading up on how to throw a general error if they're not logged in at all (except on the login page), so I can deny access to the entire application.
